src/client/datascience/jupyter/kernels/kernelDependencyService.ts

```diff
@@ -89,11 +89,17 @@ export class KernelDependencyService implements IKernelDependencyService {
             return;
         }
         if (response === KernelInterpreterDependencyResponse.selectDifferentKernel) {
-            const cmd =
-                getResourceType(resource) === 'notebook' && this.useNativeNb
-                    ? 'notebook.selectKernel'
-                    : Commands.SwitchJupyterKernel;
-            this.commandManager.executeCommand(cmd).then(noop, noop);
+            if (getResourceType(resource) === 'notebook' && this.useNativeNb) {
+                this.commandManager.executeCommand('notebook.selectKernel').then(noop, noop);
+            } else {
+                this.commandManager
+                    .executeCommand(Commands.SwitchJupyterKernel, {
```

Comment:
We 'weren't passing the right argumetns, hence kernel switching would never happen, ever.